build.assets/Dockerfile-centos7-fips

```diff
@@ -59,17 +69,17 @@ ENV GOPATH="/go" \
 
 RUN git clone https://github.com/ninja-build/ninja.git && \
     cd ninja && \
-    git checkout v1.9.0 && \
+    git checkout v1.10.2 && \
     ./configure.py --bootstrap && \
     mv ninja /usr/bin
 
 RUN mkdir -p /opt && cd /opt && \
     git clone https://github.com/google/boringssl.git && \
     cd boringssl && \
-    git checkout ae223d6138807a13006342edfeef32e813246b39 && \
+    git checkout 853ca1ea1168dff08011e5d42d94609cc0ca2e27 && \
     mkdir build && \
     cd build && \
-    scl enable llvm-toolset-7.0 "cd /opt/boringssl/build && cmake3 -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DFIPS=1 -DCMAKE_BUILD_TYPE=Release -GNinja .. && ninja"
+    scl enable llvm-toolset-13.0 "cd /opt/boringssl/build && cmake3 -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DFIPS=1 -DCMAKE_BUILD_TYPE=Release -GNinja .. && ninja"
```

Comment:
Come to think of it, I'm not sure we need to clone and build boringssl and then set `BORING_BSSL_PATH` at all.

https://docs.rs/boring/latest/boring/#support-for-pre-built-binaries-or-custom-source says:

> While this crate can build BoringSSL on its own, you may want to provide pre-built binaries instead. To do so, specify the environment variable BORING_BSSL_PATH with the path to the binaries.
>
> You can also provide specific headers by setting BORING_BSSL_INCLUDE_PATH.
>
> Notes: The crate will look for headers in the $BORING_BSSL_INCLUDE_PATH/openssl/ folder, make sure to place your headers there.
>
> In alternative a different path for the BoringSSL source code directory can be specified by setting BORING_BSSL_SOURCE_PATH which will automatically be compiled during the build process.

And then later on it says:

> Only BoringCrypto module version 853ca1ea1168dff08011e5d42d94609cc0ca2e27, as certified with [FIPS 140-2 certificate 4407](https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4407) is supported by this crate. Support is enabled by this crate’s fips feature.

So it sounds to me like we just need to update the boring dependency, ensure that the fips feature is enabled (it is), and stop pre-building.

Do you interpret that the same?

Reply:
Pinning Go was an artifact of Go pre-1.19 where you had to use an old version of Go to build/bootstrap a new version of Go that's FIPS compliant. 

Now that there's no longer a separate FIPS Go toolchain I think that's probably unnecessary. 

Plus, this Rust code has nothing to do with Go. 

+1 for relying on boring to handle this all.

Reply:
Sounds good. I was able to clean-up a fair bit of stuff. libbpf and related tools only require Clang 10+, so standardized everything on Clang 12.0.0 so that we don't need multiple Clang installations.